src/commons/achievement/AchievementView.tsx

```diff
@@ -59,11 +84,23 @@ function AchievementView(props: AchievementViewProps) {
           ))}
         </span>
       </div>
-      <AchievementViewGoal goals={goals} />
-      {status === AchievementStatus.COMPLETED && (
+
+      {status === AchievementStatus.COMPLETED &&
+        selectedAssessment &&
+        selectedAssessmentConfig &&
+        selectedAssessmentConfig.isManuallyGraded && (
+          <AchievementCommentCard courseId={courseId ?? 0} assessment={selectedAssessment} />
```

Comment:
Let's not default the `courseId` to 0 here as it might cause unintentional bugs in the future. Can we handle this properly if the course is yet to be loaded?